presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java

```diff
@@ -787,6 +788,12 @@ protected RowExpression visitSubscriptExpression(SubscriptExpression node, Void
             RowExpression base = process(node.getBase(), context);
             RowExpression index = process(node.getIndex(), context);
 
+            //this block will handle row subscript, converts the ROW_CONSTRUCTOR with subscript to a DEREFERENCE expression
+            if (base instanceof SpecialFormExpression && base.getType() instanceof RowType) {
+                ConstantExpression pos = (ConstantExpression) index;
+                int i = toIntExact((Long) pos.getValue());
```

Comment:
nit: We don't usually use local variable names such like `pos` or `i`. What about:
```java
int indexValue = toIntExact((Long) ((ConstantExpression) index).getValue());
```

Reply:
Inlined the value as anyways i need to pass a long
`if (base instanceof SpecialFormExpression && base.getType() instanceof RowType) {
                ConstantExpression pos = (ConstantExpression) index;
                return specialForm(DEREFERENCE, getType(node), base, new ConstantExpression((Long) pos.getValue() - 1, INTEGER));
            }`